### ISSUE TITLE
FolderBrowserDialog - Multiselect

### ISSUE THREAD

**Issue by Symbai** [dotnet/winforms#269]
Thanks for https://github.com/dotnet/winforms/issues/1. While the design from CommonOpenFileDialog from Windows API Codec pack has now shipped into the Folderbrowserdialog which is really amazing, sadly it does not support multiselection where CommonOpenFileDialog from the windows API codec pack does.

It would be nice to have support for selecting multiple folders, just like with CommonOpenFileDialog { IsFolderPicker = true }. It would needs a new property called SelectedPaths and a boolean to enable multi selection.

**Comment by zsd4yr** [comment]
Since the older dialog is the one used in Framework, I am adding the compat label here.

**Comment by RussKie** [comment]
To allow multi-selection a new property must be added to `FolderBrowserDialog` similar to [`OpenFileDialog.Multiselect` property](https://docs.microsoft.com/dotnet/api/system.windows.forms.openfiledialog.multiselect), and necessary plumbing to handle the new property. 
However, just adding `FOS.ALLOWMULTISELECT` to the dialog options won't work, as it doesn't appear to be supported scenario:
https://github.com/dotnet/winforms/blob/2ae45c66fdcb8e6aadf43288b7429f17f1fd0f49/src/System.Windows.Forms/src/System/Windows/Forms/FolderBrowserDialog.cs#L229

You're welcome to submit an [API proposal](https://github.com/dotnet/winforms/blob/master/Documentation/issue-guide.md#feature-and-api-suggestions) and propose an implementation, once the new API is approved.

**Comment by roland5572** [comment]
@willibrandon

I did a small test today and confirmed that after adding the FOS.ALLOWMULTISELECT flag to SetOptions . You can now select multiple folders in the dialog (the dialog UI will not display the CheckBox, but you can use the shift key to select multiple folders at the same time). Do you think it is possible to add this property too?

**Comment by Symbai** [comment]
> This issue is now marked as "up for grabs", and we’re looking for a community volunteer to work on this issue. If we receive no interest in 120 days, we will close the issue. To learn more about how we handle feature requests, please see our [documentation](https://aka.ms/winforms/issue-lifecycle).
> 
> Happy Coding!

@RussKie: So if there is no community PR, then this request is rejected. Really?!

**Comment by RussKie** [comment]
The issue has been open for 4 years now, and it doesn't appear to be of any interest to anyone else. If you feel strongly about it - you're welcome to propose a fix.

**Comment by Symbai** [comment]
Its fine if you mark it as up for grabs but who exactly gets hurt if it remains open?

**Comment by dreddy-work** [comment]
closing with active issue #8298.
